src/cpu.rs

```diff
@@ -25,32 +25,40 @@
 // ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 
-use crate::instruction::{self, DecodedInstr, Instruction, OpInput};
-use crate::memory::Memory;
+use crate::instruction::{self, AddressingMode, DecodedInstr, Instruction, OpInput};
+use crate::memory::Bus;
+
 use crate::registers::{Registers, StackPointer, Status, StatusArgs};
 
-#[derive(Clone)]
-pub struct CPU {
-    pub registers: Registers,
-    pub memory: Memory,
+fn xextend(x: u8) -> u16 {
+    u16::from(x)
 }
 
-impl Default for CPU {
-    fn default() -> Self {
-        Self::new()
-    }
+fn arr_to_addr(arr: &[u8]) -> u16 {
```

Comment:
Alternatively, we could use our own `Address { inner: u16 }` type, so we can have `Address::from(arr)`.
Just an idea, though.

Reply:
Your idea seems more idiomatic than doing `arr_to_addr` actually. 

But what about doing `u16::from(arr)` instead?

Reply:
Oh yeah, that would be even better. I was just not sure how to do that because we do some bit shifting.